File: app/src/main/java/com/dimeno/adapter/sample/adapter/UserAdapter.java
package com.dimeno.adapter.sample.adapter;

import android.os.Handler;
import android.view.ViewGroup;

import androidx.recyclerview.widget.RecyclerView;

import com.dimeno.adapter.LoadRecyclerAdapter;
import com.dimeno.adapter.annotation.LoadMoreState;
import com.dimeno.adapter.footer.LoadMoreFooter;
import com.dimeno.adapter.sample.entity.UserEntity;
import com.dimeno.adapter.sample.holder.UserViewHolder;

import java.util.ArrayList;
import java.util.List;

/**
 * UserAdapter
 * Created by wangzhen on 2020/6/9.
 */
public class UserAdapter extends LoadRecyclerAdapter<UserEntity> {

    public UserAdapter(List<UserEntity> list, ViewGroup parent) {
        super(list, parent);
    }

    @Override
    protected LoadMoreFooter createLoadMoreFooter() {
        return new UserLoadMoreFooter(this);
    }

    @Override
    public RecyclerView.ViewHolder onAbsCreateViewHolder(ViewGroup parent, int viewType) {
        return new UserViewHolder(parent);
    }

    @Override
    public void onLoadMore() {
        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                if (mDatas.size() < 15) {
                    List<UserEntity> list = new ArrayList<>();
                    for (int i = 0; i < 1; i++) {
                        list.add(new UserEntity("李四", mDatas.size() + 1));
                    }
                    addData(list);
                } else {
                    setState(LoadMoreState.NO_MORE);
                }
            }
        }, 100);
    }
}

File: app/src/main/java/com/wangzhen/adapter/sample/adapter/UserAdapter.kt
package com.wangzhen.adapter.sample.adapter

import android.os.Handler
import android.os.Looper
import android.util.Log
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.wangzhen.adapter.LoadRecyclerAdapter
import com.wangzhen.adapter.annotation.LoadMoreState
import com.wangzhen.adapter.footer.LoadMoreFooter
import com.wangzhen.adapter.sample.entity.UserEntity
import com.wangzhen.adapter.sample.holder.UserViewHolder
import java.util.*

/**
 * UserAdapter
 * Created by wangzhen on 2020/6/9.
 */
class UserAdapter(list: List<UserEntity>, parent: ViewGroup?) :
    LoadRecyclerAdapter<UserEntity>(list, parent) {
    override fun createLoadMoreFooter(): LoadMoreFooter {
        return UserLoadMoreFooter(this)
    }

    override fun onAbsCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        return UserViewHolder(parent)
    }

    override fun onLoadMore() {
        Log.e("TAG", "-> onLoadMore")
        Handler(Looper.getMainLooper()).postDelayed({
            mDatas?.let { data ->
                val list: MutableList<UserEntity> = ArrayList()
                for (i in 1..3) {
                    list.add(UserEntity("李四", data.size + i))
                }
                addData(list)
                setState(LoadMoreState.NO_MORE)
            }
        }, 1000)
    }
}